torch/fx/experimental/Partitioner.py

```diff
@@ -81,16 +93,18 @@ class Partitioner:
     along with a new graph module with partitions as submodule nodes.
     """
     def __init__(self) -> None:
-        self.partitions: Set[Partition] = set()
+        self.partitions: List[Partition] = []
         self.devices: List[Device] = []
-        self.node_to_partitions: Dict[Node, List[int]] = {}
-        self.partition_to_used_mem_bytes: Dict[Partition, int] = {}
+        self.node_to_partitions: Dict[Node, int] = {}
+        self.embedding_partition_indexes: Set[int] = set()
```

Comment:
The presence of this (sometimes unused) attribute and the `if` branches for `is_sparse_nn` lead me to believe that this class is becoming too large. Can the SparseNN partitioning strategy be expressed as a subclass of `Partitioner` that overrides the default behavior in some places? e.g. `embedding_partition_indexes` would be an attribute of the subclass only, and the `is_sparse_nn` portions of `dump_partition_DAG` and `partition_graph` would be installed via overridden methods

Reply:
I agree we should try to not pollute the base partitioner class. Another option is to contain all the scheme specific state in the method. The results from dump_partition_DAG should be in the same format regardless of what scheme is used. If we have different output fields for each partitioning scheme then the modularity of the system is compromised.